OsmAnd/src/net/osmand/plus/views/layers/MapSelectionHelper.java

```diff
@@ -457,14 +464,35 @@ private void addRoute(@NonNull MapSelectionResult result, @NonNull RotatedTileBo
 				point.x + searchRadius, point.y + searchRadius);
 		QuadRect rect = new QuadRect(minLatLon.getLongitude(), minLatLon.getLatitude(),
 				maxLatLon.getLongitude(), maxLatLon.getLatitude());
-
-		putRouteGpxToSelected(result.selectedObjects, mapLayers.getRouteSelectionLayer(), rect);
+		NetworkRouteSelectorFilter selectorFilter = new NetworkRouteSelectorFilter();
+		Set<OsmRouteType> filter = new HashSet<>();
+		List<RenderingRuleProperty> customRules = ConfigureMapUtils.getCustomRules(app,
+				UI_CATEGORY_HIDDEN, RENDERING_CATEGORY_TRANSPORT);
+		for (RenderingRuleProperty property : customRules) {
+			String attrName = property.getAttrName();
+			OsmRouteType osmRouteType = OsmRouteType.getByRenderingProperty(attrName);
+			if (osmRouteType != null) {
+				boolean enabled;
+				if (HIKING_ROUTES_OSMC_ATTR.equals(attrName)) {
+					CommonPreference<String> pref = app.getSettings().getCustomRenderProperty(attrName);
+					enabled = Arrays.asList(property.getPossibleValues()).contains(pref.get());
+				} else {
+					CommonPreference<Boolean> pref = app.getSettings().getCustomRenderBooleanProperty(attrName);
+					enabled = pref.get();
+				}
+				if (enabled) {
+					filter.add(osmRouteType);
+				}
+			}
+		}
+		selectorFilter.typeFilter = filter;
```

Comment:
Filter must be inititalize at once, otherwise we create filter for each route object!